### DISCUSSION TITLE
Adding accept to a zone-zone definition doesn't generate any rule.

### DISCUSSION THREAD

**Discussion by hpeters** [FoobarOy/foomuuri#36] [General]
localhost-public {
  accept
}
Generates the following code that doesn't work.
chain localhost-public {
                meta nfproto vmap { ipv4 : jump localhost-public_4, ipv6 : jump localhost-public_6 }
        }

        chain localhost-public_4 {
                jump allow_icmp_4
                ct state vmap { invalid : goto invalid_drop, established : accept, related : accept, new : jump smurfs_4, untracked : jump smurfs_4 }
                meta pkttype { broadcast, multicast } drop
        }

        chain localhost-public_6 {
                jump allow_icmp_6
                ct state vmap { invalid : goto invalid_drop, established : accept, related : accept, new : jump smurfs_6, untracked : jump smurfs_6 }
                meta pkttype multicast drop
        }

localhost-public {
  oifname eth0
}
Generates the code below that works.
chain localhost-public {
                meta nfproto vmap { ipv4 : jump localhost-public_4, ipv6 : jump localhost-public_6 }
        }

        chain localhost-public_4 {
                jump allow_icmp_4
                ct state vmap { invalid : goto invalid_drop, established : accept, related : accept, new : jump smurfs_4, untracked : jump smurfs_4 }
                meta pkttype { broadcast, multicast } drop
                oifname "eth0" counter packets 9 bytes 642 accept
        }

        chain localhost-public_6 {
                jump allow_icmp_6
                ct state vmap { invalid : goto invalid_drop, established : accept, related : accept, new : jump smurfs_6, untracked : jump smurfs_6 }
                meta pkttype multicast drop
                oifname "eth0" counter packets 1 bytes 99 accept
        }

**Comment by kimheino** [comment]
Can you share your full foomuur.conf file and Foomuri version number or git commit id? I tried with minimal and git head:

```
zone {
  localhost
  public
}

localhost-public {
  accept
}

public-localhost {
  reject
}
```

That generates correct ruleset:

```
chain localhost-public_4 {
  ...
  accept
}

chain public-localhost_6 {
  ...
  reject with icmpx admin-prohibited
}
```

Above is taken from Foomuuri's `next.fw` file (or `good.fw`), not from `nft list ruleset` as yours was. Can you take a look at both? I also tried loading my next.fw to nft, still fine here.

**Comment by hpeters** [comment]
The version is 0.22
Drop and reject both work.
Accept with log works but then it's logging.

I found the setting that is causing it.

foomuuri {
  counter yes
}
I had enabled counters it can be useful when debugging a firewall.
When i set counter to no then accept works.

**Comment by kimheino** [comment]
Same bug seems to be in git head too. Thanks. I'll fix it in few days.

**Comment by kimheino** [comment]
Commit 8cf1287 should fix this.

**Comment by hpeters** [comment]
Thanks.
And thank you for Foomuuri.
I've been looking for an nftables replacement for Shorewall which i have used for longer than i can remeber.
